src/generated/xinput.rs

```diff
@@ -15256,134 +15830,6 @@ impl<B: AsRef<[u8]>> TryFrom<&GenericEvent<B>> for BarrierLeaveEvent {
-        do_the_parse(&self.0).unwrap()
-    }
-    pub fn as_xproto_proximity_in(&self) -> ProximityInEvent {
-        fn do_the_parse(remaining: &[u8]) -> Result<ProximityInEvent, ParseError> {
-            let (event, remaining) = ProximityInEvent::try_parse(remaining)?;
-            let _ = remaining;
-            Ok(event)
-        }
-        do_the_parse(&self.0).unwrap()
-    }
-    pub fn as_xproto_proximity_out(&self) -> ProximityOutEvent {
-        fn do_the_parse(remaining: &[u8]) -> Result<ProximityOutEvent, ParseError> {
-            let (event, remaining) = ProximityOutEvent::try_parse(remaining)?;
-            let _ = remaining;
-            Ok(event)
-        }
-        do_the_parse(&self.0).unwrap()
-    }
-    pub fn as_xproto_device_state_notify(&self) -> DeviceStateNotifyEvent {
-        fn do_the_parse(remaining: &[u8]) -> Result<DeviceStateNotifyEvent, ParseError> {
-            let (event, remaining) = DeviceStateNotifyEvent::try_parse(remaining)?;
-            let _ = remaining;
-            Ok(event)
-        }
-        do_the_parse(&self.0).unwrap()
-    }
-    pub fn as_xproto_device_mapping_notify(&self) -> DeviceMappingNotifyEvent {
-        fn do_the_parse(remaining: &[u8]) -> Result<DeviceMappingNotifyEvent, ParseError> {
-            let (event, remaining) = DeviceMappingNotifyEvent::try_parse(remaining)?;
-            let _ = remaining;
-            Ok(event)
-        }
-        do_the_parse(&self.0).unwrap()
-    }
-    pub fn as_xproto_change_device_notify(&self) -> ChangeDeviceNotifyEvent {
-        fn do_the_parse(remaining: &[u8]) -> Result<ChangeDeviceNotifyEvent, ParseError> {
-            let (event, remaining) = ChangeDeviceNotifyEvent::try_parse(remaining)?;
-            let _ = remaining;
-            Ok(event)
-        }
-        do_the_parse(&self.0).unwrap()
-    }
-    pub fn as_xproto_device_key_state_notify(&self) -> DeviceKeyStateNotifyEvent {
-        fn do_the_parse(remaining: &[u8]) -> Result<DeviceKeyStateNotifyEvent, ParseError> {
-            let (event, remaining) = DeviceKeyStateNotifyEvent::try_parse(remaining)?;
-            let _ = remaining;
-            Ok(event)
-        }
-        do_the_parse(&self.0).unwrap()
-    }
-    pub fn as_xproto_device_button_state_notify(&self) -> DeviceButtonStateNotifyEvent {
-        fn do_the_parse(remaining: &[u8]) -> Result<DeviceButtonStateNotifyEvent, ParseError> {
-            let (event, remaining) = DeviceButtonStateNotifyEvent::try_parse(remaining)?;
-            let _ = remaining;
-            Ok(event)
-        }
-        do_the_parse(&self.0).unwrap()
-    }
-    pub fn as_xproto_device_presence_notify(&self) -> DevicePresenceNotifyEvent {
-        fn do_the_parse(remaining: &[u8]) -> Result<DevicePresenceNotifyEvent, ParseError> {
-            let (event, remaining) = DevicePresenceNotifyEvent::try_parse(remaining)?;
-            let _ = remaining;
-            Ok(event)
-        }
-        do_the_parse(&self.0).unwrap()
-    }
```

Comment:
This also removes some API on `<eventstruct>`

Reply:
Fixed